client/hosting/overview/components/migration-overview.tsx

```diff
@@ -0,0 +1,132 @@
+import { canInstallPlugins } from '@automattic/sites';
+import { translate } from 'i18n-calypso';
+import { HostingCard, HostingCardGrid } from 'calypso/components/hosting-card';
+import { HostingHero, HostingHeroButton } from 'calypso/components/hosting-hero';
+import { addQueryArgs } from 'calypso/lib/url';
+import { getMigrationStatus, getMigrationType } from 'calypso/sites-dashboard/utils';
+import type { SiteDetails } from '@automattic/data-stores';
+
+const cards = [
+	{
+		title: translate( 'Seriously secure' ),
+		text: translate(
+			'Firewalls, encryption, brute force, and DDoS protection. Your security’s all taken care of so you can stay one step ahead of any threats.'
+		),
+	},
+	{
+		title: translate( 'Unmetered bandwidth' ),
+		text: translate(
+			'With 99.999% uptime and entirely unmetered bandwidth and traffic on every plan, you’ll never need to worry about being too successful.'
+		),
+	},
+	{
+		title: translate( 'Power, meet performance' ),
+		text: translate(
+			'Our custom 28+ location CDN and 99.999% uptime ensure your site is always fast and always available from anywhere in the world.'
+		),
+	},
+	{
+		title: translate( 'Plugins, themes, and custom code' ),
+		text: translate(
+			'Build anything with full support and automatic updates for 50,000+ plugins and themes. Or start from scratch with your own custom code.'
+		),
+	},
+	{
+		title: translate( 'Expert support' ),
+		text: translate(
+			'Whenever you’re stuck, whatever you’re trying to make happen – our Happiness Engineers have the answers.'
+		),
+	},
+];
+
+const MigrationOverview = ( { site }: { site: SiteDetails } ) => {
+	if ( getMigrationStatus( site ) === 'pending' ) {
+		const migrationType = getMigrationType( site );
+
+		const baseQueryArgs = {
+			siteId: site.ID,
+			siteSlug: site.slug,
+			start: 'true',
+			ref: 'hosting-migration-overview',
+		};
+
+		let continueMigrationUrl;
+
+		if ( ! canInstallPlugins( site ) ) {
+			// For the flows where the checkout is after the choice.
+			switch ( migrationType ) {
+				case 'diy':
+					continueMigrationUrl = addQueryArgs(
+						{
+							...baseQueryArgs,
+							destination: 'upgrade',
+							how: 'myself',
+						},
+						'/setup/site-migration/site-migration-upgrade-plan'
+					);
+					break;
+				case 'difm':
+					continueMigrationUrl = addQueryArgs(
+						{
+							...baseQueryArgs,
+							destination: 'upgrade',
+							how: 'difm',
+						},
+						'/setup/site-migration/site-migration-upgrade-plan'
+					);
+					break;
+				default:
+					continueMigrationUrl = addQueryArgs(
+						baseQueryArgs,
+						'/setup/site-migration/site-migration-how-to-migrate'
+					);
+			}
+		} else {
+			// For the /setup/migration, where the checkout is before the choice.
+			switch ( migrationType ) {
+				case 'diy':
+					continueMigrationUrl = addQueryArgs(
+						baseQueryArgs,
+						'/setup/migration/migration-how-to-migrate'
```

Comment:
According to the issue, in case of `migration-pending-diy` sticker, the user should be taken to the `site-migration-instructions` step. Looks like this here is taking us to the `migration-how-to-migrate` step. In case it's not intentional, should we fix it?
<img width="803" alt="image" src="https://github.com/user-attachments/assets/c8f2e92a-c23e-4fda-8093-4f40d2a8719a">

Reply:
Good catch, @Imran92! Fixed in https://github.com/Automattic/wp-calypso/pull/95497/commits/2de7505061f8eca40813a2ca787f30d34491d2e3